### PR TITLE
Revert "Add unique_indices to jnp.take"

### PR DESCRIPTION
**Pull request by RissyRan**
Reverts google/maxtext#807 due to convergence issue (loss is nan for unknown reason)

### CODE REVIEW

**Review by ZhiyuLi-goog — APPROVED** [review]
Thank you @RissyRan ! LGTM!

**Review by gobbleturk — APPROVED** [review]
